### PR TITLE
Include GNU make in docker images to avoid issues while installing native gems

### LINKED ISSUE #81: Include GNU make in docker images to avoid issues while installing native gems
Hello there! As always, thanks for JRuby and for having nice docker images that can be used easily :)

I work at @Datadog on the [`ddtrace` gem](https://github.com/DataDog/dd-trace-rb). We support JRuby in most functionality provided by the gem, but we have an optional native extension that is MRI/CRuby-specific.

The way we make the extension "optional" is by having an `extconf.rb` that generates an effectively-empty `Makefile`. (Reference: https://github.com/DataDog/dd-trace-rb/blob/master/ext/ddtrace_profiling_native_extension/extconf.rb#L35).

I've seen this approach being used by other gems as well. `mkmf` actually includes a `dummy_makefile` method, and a few other gems use it:
* https://github.com/Shopify/gvltools/blob/master/ext/gvltools/extconf.rb#L8
* https://github.com/tmm1/stackprof/blob/v0.2.21/ext/stackprof/extconf.rb#L4
* https://github.com/Shopify/bootsnap/blob/main/ext/bootsnap/extconf.rb#L25
* [etc...](https://cs.github.com/?scopeName=All+repos&scope=&q=dummy_makefile+path%3Aextconf.rb)

Unfortunately, this approach [breaks if GNU make is not installed](https://github.com/DataDog/dd-trace-rb/issues/2244) because Rubygems will still try to call "make" and blow up since it's not available, even though the `Makefile` was an empty one just to make Rubygems happy.

**TL;DR: Would it be reasonable to include `make` by default in the docker images so that rubygems doesn't break when installing these no-op extensions**?

For context, this is what get with the current docker images:

For ddtrace:

```
root@docker-desktop:/# gem install ddtrace
Fetching ddtrace-1.5.0.gem
Fetching libddwaf-1.3.0.2.0.gem
Fetching libdatadog-0.7.0.1.1.gem
Fetching debase-ruby_core_source-0.10.17.gem
Successfully installed libddwaf-1.3.0.2.0
Successfully installed libdatadog-0.7.0.1.1
Successfully installed debase-ruby_core_source-0.10.17
Building native extensions. This could take a while...
ERROR:  Error installing ddtrace:
	ERROR: Failed to build gem native extension.

    current directory: /usr/local/bundle/gems/ddtrace-1.5.0/ext/ddtrace_profiling_native_extension
/opt/jruby/bin/jruby -I /opt/jruby/lib/ruby/stdlib -r ./siteconf20221004-195-10or6zf.rb extconf.rb

+------------------------------------------------------------------------------+
| Could not compile the Datadog Continuous Profiler because                    |
| JRuby is not supported by the Datadog Continuous Profiler.                   |
|                                                                              |
| The Datadog Continuous Profiler will not be available,                       |
| but all other ddtrace features will work fine!                               |
|                                                                              |
| Get in touch with us if you're interested in profiling your app!             |
+------------------------------------------------------------------------------+


current directory: /usr/local/bundle/gems/ddtrace-1.5.0/ext/ddtrace_profiling_native_extension
make DESTDIR\= clean
sh: 1: exec: make: not found

current directory: /usr/local/bundle/gems/ddtrace-1.5.0/ext/ddtrace_profiling_native_extension
make DESTDIR\=
sh: 1: exec: make: not found

make failed, exit code 127

Gem files will remain installed in /usr/local/bundle/gems/ddtrace-1.5.0 for inspection.
Results logged to /usr/local/bundle/extensions/universal-java-17/2.6.0/ddtrace-1.5.0/gem_make.out
```

For bootsnap:

```
root@docker-desktop:/# gem install bootsnap
Fetching msgpack-1.6.0-java.gem
Fetching bootsnap-1.13.0.gem
Successfully installed msgpack-1.6.0-java
Building native extensions. This could take a while...
ERROR:  Error installing bootsnap:
	ERROR: Failed to build gem native extension.

    current directory: /usr/local/bundle/gems/bootsnap-1.13.0/ext/bootsnap
/opt/jruby/bin/jruby -I /opt/jruby/lib/ruby/stdlib -r ./siteconf20221004-94-1cfgani.rb extconf.rb

current directory: /usr/local/bundle/gems/bootsnap-1.13.0/ext/bootsnap
make DESTDIR\= clean
sh: 1: exec: make: not found

current directory: /usr/local/bundle/gems/bootsnap-1.13.0/ext/bootsnap
make DESTDIR\=
sh: 1: exec: make: not found

make failed, exit code 127

Gem files will remain installed in /usr/local/bundle/gems/bootsnap-1.13.0 for inspection.
Results logged to /usr/local/bundle/extensions/universal-java-17/2.6.0/bootsnap-1.13.0/gem_make.out
```

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**:

This PR adds GNU make to the `apt-get install` set of all `Dockerfile`s.

**Motivation**:

As discussed in #81, several gems (`ddtrace`, `bootsnap`, `stackprof`, `gvltools`) autodetect in their `extconf.rb` if they're running on JRuby, and avoid attempting to compile their native extension in that case.

Unfortunately, the "avoid attempting to compile" is implemented by writing a dummy almost-empty `Makefile`, which rubygems still tries to run. This breaks if GNU make is not installed.

GNU make is quite small, so this seems a nice improvement in user experience for users of the JRuby docker images.

**How to test the change?**:

Before:

```
$ gem install bootsnap
Fetching bootsnap-1.14.0.gem
Fetching msgpack-1.6.0-java.gem
Successfully installed msgpack-1.6.0-java
Building native extensions. This could take a while...
ERROR:  Error installing bootsnap:
	ERROR: Failed to build gem native extension.

    current directory: /usr/local/bundle/gems/bootsnap-1.14.0/ext/bootsnap
/opt/jruby/bin/jruby -I /opt/jruby/lib/ruby/stdlib -r ./siteconf20221121-34-we57cb.rb extconf.rb

current directory: /usr/local/bundle/gems/bootsnap-1.14.0/ext/bootsnap
make DESTDIR\= clean
sh: 1: exec: make: not found

current directory: /usr/local/bundle/gems/bootsnap-1.14.0/ext/bootsnap
make DESTDIR\=
sh: 1: exec: make: not found

make failed, exit code 127

Gem files will remain installed in /usr/local/bundle/gems/bootsnap-1.14.0 for inspection.
Results logged to /usr/local/bundle/extensions/universal-java-17/2.6.0/bootsnap-1.14.0/gem_make.out
```

After:

```
$ gem install bootsnap
Fetching msgpack-1.6.0-java.gem
Fetching bootsnap-1.14.0.gem
Successfully installed msgpack-1.6.0-java
Building native extensions. This could take a while...
Successfully installed bootsnap-1.14.0
2 gems installed
```

---

Fixes #81

### CODE REVIEW

**Comment by ivoanjo** [comment]
Update: Rebased PR and added makefile to the newly-added 9.4 `Dockerfile`s as well. Congrats on the 9.4 release, btw!

**Comment by headius** [comment]
Yup good, I missed this before the holidays but I'll merge now and update Hub!
